core/src/main/java/org/jdbi/v3/core/array/SqlArrayTypes.java

```diff
@@ -72,7 +73,8 @@ public SqlArrayTypes setArgumentStrategy(SqlArrayArgumentStrategy argumentStrate
      * @return this
      */
     public SqlArrayTypes register(Class<?> elementType, String sqlTypeName) {
-        return register(VendorSupportedArrayType.factory(elementType, sqlTypeName));
+        final Optional<SqlArrayType<?>> arrayType = Optional.of(SqlArrayType.of(sqlTypeName, Function.identity()));
```

Comment:
Shouldn't we use `SqlArrayTypeFactory.of()` here instead of duplicating the type check?